haeun/RestApiStudy/src/main/java/com/example/restapistudy/repository/MemoryBoardRepository.java

```diff
@@ -0,0 +1,42 @@
+package com.example.restapistudy.repository;
+
+import com.example.restapistudy.domain.Board;
+import org.springframework.stereotype.Repository;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// BoardRepository 인터페이스 구현체
+@Repository
+public class MemoryBoardRepository implements BoardRepository{
+
+    // 글 목록들을 저장할 컬렉션
+    private static Map<Long,Board> store = new HashMap<>();
+
+    @Override
+    public void save(Board board) {
+        store.put(board.getId(), board);
+    }
+
+    @Override
+    public Board findById(Long id) {
+       return store.get(id);
+    }
+
+    @Override
+    public List<Board> findAll() { // 최신순으로 정렬
+        return store.values().stream().sorted(Comparator.comparing(Board::getDate).reversed()).toList();
+    }
```

Comment:
```suggestion
        @Override
        public List<Board> findAll() { // 최신순으로 정렬
            return store.values().stream()
                    .sorted(Comparator.comparing(Board::getDate)
                    .reversed())
                    .toList();
        }
```

이렇게 줄바꿈을 통해서 작성하시면 코드를 읽기 더 좋을 것 같아요!

Reply:
오호오 그렇네요, 감사합니다!

Reply:
디미터의 법칙에 대해서 학습해보셔도 좋을 것 같습니다